homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,179 @@
+                    )
+                    residences = acct.get_residences()
+                    for res in residences:
+                        switches = res.get_iot_switches()
+                        for switch in switches:
+                            # Add the switch to the appropriate list in the iot_switches dictionary.
+                            platform = DecoraWifiPlatform.classifydevice(switch)
+                            self._iot_switches[platform].append(switch)
+                elif permission.residenceId is not None:
+                    residence = Residence(self._session, permission.residenceId)
+                    switches = residence.get_iot_switches()
+                    for switch in switches:
+                        # Add the switch to the appropriate list in the iot_switches dictionary.
+                        platform = DecoraWifiPlatform.classifydevice(switch)
+                        self._iot_switches[platform].append(switch)
+        except ValueError as exc:
+            raise CommFailed from exc
+
+    def reauth(self):
+        """Reauthenticate this object's session."""
+        self._api_logout()
+        self._session = DecoraWiFiSession()
+        self._api_login()
+
+    def refresh_devices(self):
+        """Refresh this object's devices."""
+        self._iot_switches: dict[str, IotSwitch] = {
+            platform: [] for platform in PLATFORMS
+        }
+        self._api_get_devices()
+
+    def setup(self):
+        """Set up the session after object instantiation."""
+        self._api_login()
+        self._api_get_devices()
+
+    def teardown(self):
+        """Clean up the session on object deletion."""
+        self._api_logout()
+
+    @staticmethod
+    async def async_setup_decora_wifi(hass: HomeAssistant, email: str, password: str):
+        """Set up a decora wifi session."""
+
+        def setupplatform() -> DecoraWifiPlatform:
+            platform = DecoraWifiPlatform(email, password)
+            platform.setup()
+            return platform
+
+        return await hass.async_add_executor_job(setupplatform)
+
+    @staticmethod
+    def classifydevice(dev):
+        """Classify devices by platform."""
+        # The light platform is the only one currently implemented in the integration.
+        return LIGHT_DOMAIN
+
+
+class DecoraWifiEntity(Entity):
+    """Base Class for decora_wifi entities."""
+
+    def __init__(self, device: IotSwitch) -> None:
+        """Initialize Decora Wifi device base class."""
+        self._switch = device
+        self._model = device.model
+        self._unique_id = device.mac
+
+    @property
+    def device_info(self):
+        """Return device info for the associated device."""
+        return {
+            "name": self._switch.name,
+            "connections": {(device_registry.CONNECTION_NETWORK_MAC, self._unique_id)},
+            "identifiers": {(DOMAIN, self._unique_id)},
```

Comment:
```suggestion
```

We only need one or the other.  Since we have the mac, that is prefered

Reply:
This is a tuple containing `DOMAIN` and the MAC address, per [this documentation](https://developers.home-assistant.io/docs/device_registry_index/#defining-devices).  I think the purpose is to ensure we're not assuming a device's unique id remains unique outside of `DOMAIN`--seemed like a reasonable precaution even though a MAC technically *should* be globally unique.  But I can definitely toss the tuple and just use `self._unique_id` if you feel it's not needed.

Reply:
Strike that, I think I misunderstood what you were saying.  You're telling me I don't need to provide an `identifiers` key because I've already provided the `connections` key, correct?

Reply:
Correct